src/chains/algorand_1/algoTransaction.ts

```diff
@@ -267,25 +299,51 @@ export class AlgorandTransaction implements Transaction {
     }
   }
 
-  /** Returns address, for which, a matching signature must be attached to transaction
-   *  Throws if actions[].from is not a valid address - needed to determine the required signature */
+  /** Returns address, for which, a matching signature must be attached to transaction */
   public get missingSignatures(): AlgorandAddress[] {
     this.assertIsValidated()
-    const missingSignature = this.hasAllRequiredSignatures ? null : this.requiredAuthorization
-    return isNullOrEmpty(missingSignature) ? null : [missingSignature] // if no values, return null instead of empty array
+    const missingSignatures = this.requiredAuthorizations?.filter(
+      auth => !this.hasSignatureForPublicKey(getAlgorandPublicKeyFromAddress(auth)),
+    )
+
+    /** check if number of signatures present are greater then or equal to multisig threshold.
+     * If so, set missing signatures to null */
+    if (this.multiSigOptions) {
+      return this.multiSigOptions.addrs.length - (this.multiSigOptions.threshold as number) <= missingSignatures.length
```

Comment:
dont need to cast a number as number

Reply:
done. Threshold was typed as "Number" instead of "number"

Reply:
done